sqrl-planner/src/main/java/com/datasqrl/engine/stream/flink/plan/SqrlToFlinkSqlGenerator.java

```diff
@@ -379,23 +380,22 @@ private SqlCreateTable registerSinkTable(WriteSink sink, RelNode relNode,
       EngineSink engineSink = (EngineSink) sink;
       TableConfig engineConfig = engineSink.getStage().getEngine().getSinkConfig(engineSink.getNameId());
 
-      Optional<StagePlan> stagePlan = stagePlans.stream()
+      //Add partitioning keys if engine supports them
+      StagePlan stagePlan = stagePlans.stream()
           .filter(f -> f.getStage() == engineSink.getStage())
-          .findFirst();
+          .findFirst().orElseThrow();
       TableConfigBuilder builder = engineConfig.toBuilder();
-      if (stagePlan.isPresent()) {
-        StagePlan stagePlan1 = stagePlan.get();
-        if (isPartitionable(stagePlan1.getStage().getName())) {
-          DatabaseStagePlan dbPlan = (DatabaseStagePlan) stagePlan1;
-          String tableId = engineSink.getNameId();
-          Optional<IndexDefinition> optIndex =  dbPlan.getIndexDefinitions().stream()
-              .filter(i -> i.getTableId().equals(tableId))
-              .findFirst();
-          if (optIndex.isPresent()) {
-            IndexDefinition mainIndex = optIndex.get();
-            builder.setPartitionKey(mainIndex.getColumnNames());
-          }
-
+      if (engineSink.getStage().supportsFeature(EngineFeature.PARTITIONING)) {
+        DatabaseStagePlan dbPlan = (DatabaseStagePlan) stagePlan;
+        String tableId = engineSink.getNameId();
+        Optional<IndexDefinition> optIndex =  dbPlan.getIndexDefinitions().stream()
+            .filter(idx -> idx.getTableId().equals(tableId))
+            .filter(idx -> idx.getType().isPartitioned())
+            .findFirst();
+        if (optIndex.isPresent()) {
+          IndexDefinition partitionIndex = optIndex.get();
+          List<String> partitionColumns = partitionIndex.getColumnNames().subList(0, partitionIndex.getPartitionOffset());
```

Comment:
Huh, this could be deceptive from an interface perspective.

Reply:
What would be a way to make this more obvious? 
A partitioned btree is defined by a set of columns for the partition followed by a set of columns for the sort order.
In many systems (e.g. Cassandra, Citrix) that's indicated by an offset.